detox/src/devices/Device.js

```diff
@@ -44,11 +44,18 @@ class Device {
   }
 
   async setOrientation(orientation) {
-    // orientation is 'landscape' (meaning left side portrait) or 'portrait' (non-reversed)
-    const call = invoke.call(
-      invoke.IOS.EarlGrey(''),
+    // keys are possible orientations
+    const orientationMapping = {
+      landscape: 3, // top at left side landscape
+      portrait: 1  // non-reversed portrait
+    };
+    if (!Object.keys(orientationMapping).includes(orientation)) {
+      throw new Error(`setOrientation failed: provided orientation ${orientation} is not part of supported orientations: ${Object.keys(orientationMapping)}`)
+    }
+
+    const call = invoke.EarlGrey.call(
```

Comment:
take a look here
```js
this._call = invoke.call(invoke.EarlGrey.instance, 'detox_selectElementWithMatcher:', matcher._call);
```
https://github.com/wix/detox/blob/master/detox/src/ios/expect.js#L229

This function calls the static method`detox_selectElementWithMatcher`  function in `EarlGrey`